src/context.ts

```diff
@@ -388,6 +388,42 @@ export class Context implements RenamedUpdate {
                 this.chosenInlineResult?.inline_message_id
         );
     }
+    /**
+     * Get entities and their text. Extracts the text from `ctx.msg.text` or `ctx.msg.caption`.
+     * Returns an empty array if one of `ctx.msg`, `ctx.msg.text`
+     * or `ctx.msg.entities` is undefined.
+     *
+     * You can filter specific entity types by passing the `types` parameter. Example:
+     *
+     * ```ts
+     * ctx.entities() // Returns all entity types
+     * ctx.entities('url') // Returns only url entities
+     * ctx.enttities(['url', 'email']) // Returns url and email entities
+     * ```
+     *
+     * @param types Types of entities to return. Omit to get all entities.
+     * @returns Array of entities and their texts, or empty array when there's no text
+     */
+    entities(
+        types?: MaybeArray<MessageEntity["type"]>,
+    ): Array<MessageEntity & { text: string }> {
```

Comment:
Could we enforce the correct return type based on the parameter? For example, it would be cool if passing `"text_link"` would make sure that the `url` property is always set.

Reply:
I'll give it a try :)